libs/api/windows_platform.cpp

```diff
@@ -13,83 +13,60 @@
     size_t size,
     const struct ebpf_platform_t*,
-    ebpf_verifier_options_t options)
+    ebpf_verifier_options_t)
 {
-    if (size % sizeof(ebpf_map_definition_t) != 0) {
+    if (size % sizeof(ebpf_map_definition_in_file_t) != 0) {
         throw std::runtime_error(
             std::string("bad maps section size, must be a multiple of ") +
-            std::to_string(sizeof(ebpf_map_definition_t)));
+            std::to_string(sizeof(ebpf_map_definition_in_file_t)));
     }
 
-    auto mapdefs =
-        std::vector<ebpf_map_definition_t>((ebpf_map_definition_t*)data, (ebpf_map_definition_t*)(data + size));
+    auto mapdefs = std::vector<ebpf_map_definition_in_file_t>(
+        (ebpf_map_definition_in_file_t*)data, (ebpf_map_definition_in_file_t*)(data + size));
     for (int i = 0; i < mapdefs.size(); i++) {
         auto& s = mapdefs[i];
-        map_descriptors.emplace_back(EbpfMapDescriptor{
-            .original_fd = create_map_windows(
-                s.type,
-                s.key_size,
-                s.value_size,
-                s.max_entries,
-                s.inner_map_idx,
-                (i * sizeof(ebpf_map_definition_t)),
-                options),
+        uint32_t section_offset = (i * sizeof(ebpf_map_definition_in_file_t));
+
+        // The map file descriptors that appear in eBPF bytecode start at 1,
+        // in the order the maps appear in the maps section.
+        const int ORIGINAL_FD_OFFSET = 1;
+
+        int original_fd = i + ORIGINAL_FD_OFFSET;
+        unsigned int inner_map_original_fd = UINT_MAX;
+        if (s.type == BPF_MAP_TYPE_ARRAY_OF_MAPS || s.type == BPF_MAP_TYPE_HASH_OF_MAPS) {
+            inner_map_original_fd = (unsigned int)s.inner_map_idx + ORIGINAL_FD_OFFSET;
+        }
+
+        cache_map_handle(
+            INVALID_HANDLE_VALUE,
+            original_fd,
+            s.type,
+            s.key_size,
+            s.value_size,
+            s.max_entries,
+            inner_map_original_fd,
+            section_offset);
+
+        verifier_map_descriptors.emplace_back(EbpfMapDescriptor{
+            .original_fd = original_fd,
             .type = (uint32_t)s.type,
             .key_size = s.key_size,
             .value_size = s.value_size,
-            .inner_map_fd = s.inner_map_idx});
-    }
-    for (size_t i = 0; i < mapdefs.size(); i++) {
-        unsigned int inner = mapdefs[i].inner_map_idx;
-        if (inner >= map_descriptors.size())
-            throw std::runtime_error(
-                std::string("bad inner map index ") + std::to_string(inner) + " for map " + std::to_string(i));
-        map_descriptors[i].inner_map_fd = map_descriptors.at(inner).original_fd;
+            .inner_map_fd = inner_map_original_fd});
     }
 }
 
 const ebpf_platform_t g_ebpf_platform_windows = {
     get_program_type_windows,
     get_helper_prototype_windows,
     is_helper_usable_windows,
-    sizeof(ebpf_map_definition_t),
+    sizeof(ebpf_map_definition_in_memory_t),
```

Comment:
Should this be `ebpf_map_definition_in_file_t`?

Reply:
Yes thanks.  (Though at present they're the same size.)